src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php

```diff
@@ -21,23 +24,39 @@
  */
 final class SesTransportFactory extends AbstractTransportFactory
 {
+    private $credentialProvider;
+
+    public function __construct()
+    {
+        $this->credentialProvider = new InstanceCredentialProvider();
+    }
```

Comment:
You must call the parent constructor (and fix the constructor arguments accordingly). That will allow to inject the HTTP client as well to `InstanceCredentialProvider`.